### ISSUE TITLE
Disabling blocks doesn't disable variables

### ISSUE THREAD

**Issue by CGracey** [parallaxinc/BlocklyProp#336]
This issue is fairly low priority since it doesn't actually stop the code from successfully compiling or running. I was testing all the MX2125 with five different variables and I disabled a few blocks where variables where defined, but they still appeared in the code as integers despite them being disabled. It may be helpful to remove variables in the code if the only blocks they are used in are disabled.

![tilt](https://cloud.githubusercontent.com/assets/20099916/16739273/2bf7beaa-474e-11e6-9fa2-c2f92a609779.PNG)

![rotate](https://cloud.githubusercontent.com/assets/20099916/16739279/2f487e0a-474e-11e6-9eb0-7201525327ed.PNG)

![acceleration](https://cloud.githubusercontent.com/assets/20099916/16739287/379f84c2-474e-11e6-8fc6-601b2b1b8227.PNG)

``` C
/* SERIAL_TERMINAL USED */
#include "simpletools.h"
#include "mx2125.h"

int X_Tilt;
int Rotate;

int X_accel;

int Y_Tilt;

int Y_accel;


int main()
{

    while (1)
    {
        X_accel = (mx_accel(10));
        Y_accel = (mx_accel(11));
        print("X_accel = %d\n", X_accel);
        print("Y_accel = %d\n", Y_accel);
        pause(250);
    }

}
```

Would turn into:

``` C
/* SERIAL_TERMINAL USED */
#include "simpletools.h"
#include "mx2125.h"

int X_accel;


int Y_accel;


int main()
{

    while (1)
    {
        X_accel = (mx_accel(10));
        Y_accel = (mx_accel(11));
        print("X_accel = %d\n", X_accel);
        print("Y_accel = %d\n", Y_accel);
        pause(250);
    }

}
```

**Comment by DarkmatterVale** [comment]
hmmm I see what you mean. I will have to look into how disabled blocks work...

**Comment by michel-cf** [comment]
When I update to the new Blockly this might solve it: https://github.com/google/blockly/pull/472
(We'll need to see what the actual impact of this is)

**Comment by DarkmatterVale** [comment]
I'm gonna hand this over to @michel-cf because this is more of an internal blocky issue I believe.

**Comment by MatzElectronics** [comment]
@csavage-parallax @mrodriguez-parallax could one of you re-test this?

**Comment by csavage-parallax** [comment]
Miguel can test this one. ;)


*Chris Savage | *Engineering Tech | Main Office: (916) 624-8333 | Direct to
Tech Support: (888) 997-8267 | www.parallax.com |

**Comment by mrodriguez-parallax** [comment]
Created 3 different variables for 3 pings. The variables are value1, value2, and value3.  I disabled value1 and value2 along with their supporting blocks as shown below.

![image](https://cloud.githubusercontent.com/assets/20686764/21114167/08acbda6-c061-11e6-99a7-f19a539eff38.png)


In the actual text code value1 and value2 are still initialized as ints. The variables were removed from the main code block when disabled but are still present in the initialization section.
![image](https://cloud.githubusercontent.com/assets/20686764/21114279/6dae73f2-c061-11e6-8f71-6a9b154594d0.png)

**Comment by paragitadmin** [comment]
Has anyone done the analysis of what happens in the case where multiple blocks reference a variable and at least one of those blocks is not disabled?

**Comment by MatzElectronics** [comment]
I'd like to close this - this is at the Blockly Core, and a little beyond our collective expertise - and it has no impact on functionality or performance.  Any objections?

**Comment by mrodriguez-parallax** [comment]
No objections from me. If it doesn't affect performance or output any errors I don't see it can't wait or even close out entirely.
miguel 

    On Monday, January 16, 2017 11:59 PM, Matthew Matz <notifications@github.com> wrote:
 

 I'd like to close this - this is at the Blockly Core, and a little beyond our collective expertise - and it has no impact on functionality or performance. Any objections?—

**Comment by MatzElectronics** [comment]
I now know how to accomplish this, but it may take me awhile to get there...  Within the generator for any block that uses variables, it has to first call a "isDisabled()", and if so, not add the variable names to the variableDB/setup code.

**Comment by MatzElectronics** [comment]
I thought I had this figured out...and I don't.  I do believe it's possible, but I don't believe it's worth the mass of code it will take.  Closing.

**Comment by zfi** [comment]
@MatzElectronics It appears that this has been addressed in the Blockly core.

https://github.com/google/blockly/commit/97801a0e2fe1a5130b007ed6ab8bfd879a7be56e

Seems that is a block is disabled, the call to AllUsedVariables() should return a list that excludes variables associated with the disabled blocks.

**Comment by MatzElectronics** [comment]
Okay - I'll investigate this.

**Comment by paragitadmin** [comment]
Issue corrected in PR #1118. Closing issue.
